### ISSUE TITLE
ValueError in `numpy.concatenate` during active labeling in Record Linkage and Gazeteer examples

### ISSUE THREAD

**Issue by manusturla** [dedupeio/dedupe#1108]
Hi! I'm having some issues when running Record Linkage and Gazeteer examples.

During **active labeling**, after 2 responses (it does not matter if I respond 'yes', 'no' or 'unsure') it raises a ValueError:
> ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 0, the array at index 0 has size 4998 and the array at index 1 has size 4996

### Tracebacks

[Record Linkage](https://github.com/dedupeio/dedupe-examples/tree/master/record_linkage_example):
```
Traceback (most recent call last):
  File "/mnt/74225F01225EC82E/Archivos/FIUBA/Trabajo Profesional/dedupe_examples_[exclude]/record_linkage_example/record_linkage_example.py", line 140, in <module>
    dedupe.console_label(linker)
  File "/mnt/74225F01225EC82E/Archivos/FIUBA/Trabajo Profesional/dedupe_examples_[exclude]/record_linkage_example/venv/lib/python3.10/site-packages/dedupe/convenience.py", line 150, in console_label
    unlabeled = deduper.uncertain_pairs()
  File "/mnt/74225F01225EC82E/Archivos/FIUBA/Trabajo Profesional/dedupe_examples_[exclude]/record_linkage_example/venv/lib/python3.10/site-packages/dedupe/api.py", line 1168, in uncertain_pairs
    return [self.active_learner.pop()]
  File "/mnt/74225F01225EC82E/Archivos/FIUBA/Trabajo Profesional/dedupe_examples_[exclude]/record_linkage_example/venv/lib/python3.10/site-packages/dedupe/labeler.py", line 332, in pop
    probs = numpy.concatenate(prob_l, axis=1)
  File "<__array_function__ internals>", line 180, in concatenate
ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 0, the array at index 0 has size 4998 and the array at index 1 has size 4996
```

[Gazeteer](https://github.com/dedupeio/dedupe-examples/tree/master/gazetteer_example):
```
Traceback (most recent call last):
  File "/mnt/74225F01225EC82E/Archivos/FIUBA/Trabajo Profesional/dedupe_examples_[exclude]/gazetteer_example/gazetteer_example.py", line 140, in <module>
    dedupe.console_label(gazetteer)
  File "/home/manfer/.local/lib/python3.10/site-packages/dedupe/convenience.py", line 150, in console_label
    unlabeled = deduper.uncertain_pairs()
  File "/home/manfer/.local/lib/python3.10/site-packages/dedupe/api.py", line 1168, in uncertain_pairs
    return [self.active_learner.pop()]
  File "/home/manfer/.local/lib/python3.10/site-packages/dedupe/labeler.py", line 331, in pop
    probs = numpy.concatenate(prob_l, axis=1)
  File "<__array_function__ internals>", line 180, in concatenate
ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 0, the array at index 0 has size 4998 and the array at index 1 has size 4996
```
### Environment
Python version: `Python 3.10.6`.

#### Packages installed in virtual environment
For Record Linkage:
```
Package                  Version
------------------------ --------
affinegap                1.12
BTrees                   4.10.1
categorical-distance     1.9
cffi                     1.15.1
datetime-distance        0.1.3
dedupe                   2.0.18
dedupe-variable-datetime 0.1.5
DoubleMetaphone          1.1
future                   0.18.2
haversine                2.7.0
highered                 0.2.1
joblib                   1.2.0
Levenshtein-search       1.4.5
numpy                    1.23.3
persistent               4.9.1
pip                      22.2.2
pycparser                2.21
pyhacrf-datamade         0.2.6
PyLBFGS                  0.2.0.14
python-dateutil          2.8.2
scikit-learn             1.1.2
scipy                    1.9.1
setuptools               65.3.0
simplecosine             1.2
six                      1.16.0
threadpoolctl            3.1.0
typing_extensions        4.3.0
Unidecode                1.3.6
wheel                    0.37.1
zope.index               5.2.1
zope.interface           5.4.0
```

For Gazeteer:
```
Package                  Version
------------------------ ---------------
affinegap                1.12
alembic                  1.7.7
apispec                  5.2.2
apparmor                 3.0.4
appdirs                  1.4.4
APScheduler              3.8.1
apturl                   0.5.2
attrs                    21.4.0
autopep8                 1.7.0
bcrypt                   3.2.0
beautifulsoup4           4.11.1
black                    20.8b1
blinker                  1.4
boto3                    1.18.12
botocore                 1.21.65
Brlapi                   0.8.3
bs4                      0.0.1
BTrees                   4.10.1
categorical-distance     1.9
certifi                  2020.6.20
cffi                     1.15.1
cfgv                     3.3.1
chardet                  4.0.0
charset-normalizer       2.0.12
click                    8.0.3
colorama                 0.4.4
command-not-found        0.3
coverage                 5.3
cPython                  0.0.6
cryptography             3.4.7
cssutils                 2.6.0
cupshelpers              1.0
currency-symbols         2.0.3
datetime-distance        0.1.3
dbus-python              1.2.18
dedupe                   2.0.18
dedupe-variable-datetime 0.1.5
defer                    1.0.6
distlib                  0.3.5
distro                   1.7.0
distro-info              1.1build1
dnspython                2.2.1
DoubleMetaphone          1.1
duplicity                0.8.21
email-validator          1.1.1
et-xmlfile               1.0.1
fasteners                0.14.1
filelock                 3.8.0
flake8                   3.8.4
Flask                    2.0.2
flask-apispec            0.11.4
Flask-APScheduler        1.12.3
Flask-Bcrypt             0.7.1
Flask-Cors               3.0.10
Flask-JWT                0.2.0
Flask-Mail               0.9.0
flask-marshmallow        0.14.0
Flask-Migrate            2.6.0
Flask-Script             2.0.6
Flask-SQLAlchemy         2.5.1
fonttools                4.37.1
freezegun                1.1.0
future                   0.18.2
fuzzywuzzy               0.18.0
glob2                    0.7
greenlet                 1.1.2
gunicorn                 20.0.4
haversine                2.7.0
highered                 0.2.1
html5lib                 1.1
httplib2                 0.20.2
identify                 2.5.3
idna                     3.3
importlib-metadata       4.6.4
iniconfig                1.1.1
itsdangerous             2.1.2
jdcal                    1.4.1
jeepney                  0.7.1
Jinja2                   3.1.2
jmespath                 0.10.0
joblib                   1.2.0
keyring                  23.5.0
language-selector        0.1
launchpadlib             1.10.16
lazr.restfulclient       0.14.4
lazr.uri                 1.0.6
Levenshtein-search       1.4.5
LibAppArmor              3.0.4
libvirt-python           8.0.0
lockfile                 0.12.2
louis                    3.20.0
lxml                     4.9.1
macaroonbakery           1.3.1
Mako                     1.1.3
MarkupSafe               2.1.1
marshmallow              3.15.0
marshmallow-sqlalchemy   0.24.1
mccabe                   0.6.1
meson                    0.61.2
monotonic                1.6
more-itertools           8.10.0
mypy-extensions          0.4.3
netifaces                0.11.0
nodeenv                  1.7.0
numpy                    1.23.2
oauthlib                 3.2.0
olefile                  0.46
openpyxl                 3.0.3
openpyxl-dictreader      0.1.3
packaging                21.3
pandas                   1.2.3
paramiko                 2.9.3
parse                    1.19.0
parse-type               0.6.0
pathspec                 0.9.0
persistent               4.9.1
pexpect                  4.8.0
Pillow                   8.4.0
pip                      22.0.2
platformdirs             2.5.2
pluggy                   0.13.1
pre-commit               2.10.1
protobuf                 3.12.4
psycopg2                 2.8
ptyprocess               0.7.0
py                       1.11.0
PyAudio                  0.2.11
pycairo                  1.20.1
pycodestyle              2.6.0
pycparser                2.21
pycups                   2.0.1
pyflakes                 2.2.0
Pygments                 2.11.2
PyGObject                3.42.1
pyhacrf-datamade         0.2.6
PyJWT                    2.3.0
PyLBFGS                  0.2.0.14
pymacaroons              0.13.0
pymongo                  4.2.0
PyMySQL                  1.0.2
PyNaCl                   1.5.0
pyparsing                2.4.7
pyRFC3339                1.1
pytest                   7.1.2
pytest-bdd               6.0.1
pytest-cov               2.10.1
python-apt               2.3.0+ubuntu2.1
python-barcode           0.13.1
python-dateutil          2.8.1
python-debian            0.1.43ubuntu1
python-dotenv            0.15.0
python-Levenshtein       0.12.2
pytz                     2022.1
pytz-deprecation-shim    0.1.0.post0
pyxdg                    0.27
pyxlsb                   1.0.9
PyYAML                   5.4.1
regex                    2022.8.17
reportlab                3.6.8
requests                 2.26.0
s3transfer               0.5.2
scikit-learn             1.1.2
scipy                    1.9.1
screen-resolution-extra  0.0.0
SecretStorage            3.3.1
setuptools               59.6.0
simplecosine             1.2
six                      1.16.0
soupsieve                2.3.2.post1
SQLAlchemy               1.4.41
systemd-python           234
threadpoolctl            3.1.0
toml                     0.10.2
tomli                    2.0.1
twilio                   7.3.1
typed-ast                1.5.4
typing_extensions        4.3.0
tzdata                   2022.1
tzlocal                  4.2
ubuntu-advantage-tools   27.10
ubuntu-drivers-common    0.0.0
ufw                      0.36.1
unattended-upgrades      0.1
Unidecode                1.3.6
urllib3                  1.26.5
usb-creator              0.3.7
virtualenv               20.16.3
wadllib                  1.3.6
wajig                    4.0.3
webargs                  8.2.0
webcolors                1.12
webencodings             0.5.1
websocket-client         0.59.0
Werkzeug                 2.0.3
wheel                    0.37.1
xdg                      5
xkit                     0.0.0
xlrd                     2.0.1
xls2xlsx                 0.1.5
XlsxWriter               3.0.3
xlwings                  0.27.14
zipp                     1.0.0
zope.index               5.2.1
zope.interface           5.4.0
```

**Comment by fgregg** [comment]
i've actually fixed this on main. let me cut a release.

**Comment by fgregg** [comment]
should be resolved in dedupe 2.0.19
